sympy/physics/continuum_mechanics/beam.py

```diff
@@ -710,141 +769,6 @@ def applied_loads(self):
-                elif load[2] > 0:
-                    load_2 -= load[0]*SingularityFunction(x, load[3] - l, load[2]) + load[0]*SingularityFunction(x, load[3] - l, 0)
-
-        h = Symbol('h')     # Force due to hinge
-        load_1 += h*SingularityFunction(x, l, -1)
-        load_2 -= h*SingularityFunction(x, 0, -1)
-
-        eq = []
-        shear_1 = integrate(load_1, x)
-        shear_curve_1 = limit(shear_1, x, l)
-        eq.append(shear_curve_1)
-        bending_1 = integrate(shear_1, x)
-        moment_curve_1 = limit(bending_1, x, l)
-        eq.append(moment_curve_1)
-
-        shear_2 = integrate(load_2, x)
-        shear_curve_2 = limit(shear_2, x, self.length - l)
-        eq.append(shear_curve_2)
-        bending_2 = integrate(shear_2, x)
-        moment_curve_2 = limit(bending_2, x, self.length - l)
-        eq.append(moment_curve_2)
-
-        C1 = Symbol('C1')
-        C2 = Symbol('C2')
-        C3 = Symbol('C3')
-        C4 = Symbol('C4')
-        slope_1 = S.One/(E*I1)*(integrate(bending_1, x) + C1)
-        def_1 = S.One/(E*I1)*(integrate((E*I)*slope_1, x) + C1*x + C2)
-        slope_2 = S.One/(E*I2)*(integrate(integrate(integrate(load_2, x), x), x) + C3)
-        def_2 = S.One/(E*I2)*(integrate((E*I)*slope_2, x) + C4)
-
-        for position, value in self.bc_slope:
-            if position<l:
-                eq.append(slope_1.subs(x, position) - value)
-            else:
-                eq.append(slope_2.subs(x, position - l) - value)
-
-        for position, value in self.bc_deflection:
-            if position<l:
-                eq.append(def_1.subs(x, position) - value)
-            else:
-                eq.append(def_2.subs(x, position - l) - value)
-
-        eq.append(def_1.subs(x, l) - def_2.subs(x, 0)) # Deflection of both the segments at hinge would be equal
-
-        constants = list(linsolve(eq, C1, C2, C3, C4, h, *reactions))
-        reaction_values = list(constants[0])[5:]
-
-        self._reaction_loads = dict(zip(reactions, reaction_values))
-        self._load = self._load.subs(self._reaction_loads)
-
-        # Substituting constants and reactional load and moments with their corresponding values
-        slope_1 = slope_1.subs({C1: constants[0][0], h:constants[0][4]}).subs(self._reaction_loads)
-        def_1 = def_1.subs({C1: constants[0][0], C2: constants[0][1], h:constants[0][4]}).subs(self._reaction_loads)
-        slope_2 = slope_2.subs({x: x-l, C3: constants[0][2], h:constants[0][4]}).subs(self._reaction_loads)
-        def_2 = def_2.subs({x: x-l,C3: constants[0][2], C4: constants[0][3], h:constants[0][4]}).subs(self._reaction_loads)
-
-        self._hinge_beam_slope = slope_1*SingularityFunction(x, 0, 0) - slope_1*SingularityFunction(x, l, 0) + slope_2*SingularityFunction(x, l, 0)
-        self._hinge_beam_deflection = def_1*SingularityFunction(x, 0, 0) - def_1*SingularityFunction(x, l, 0) + def_2*SingularityFunction(x, l, 0)
-
```

Comment:
It looks like you are removing the prior functionality in favor of the new way. If this changes any thing that is public and user facing, then we have to follow the deprecation policy. Does this change anything for the user?

Also, does swapping for the new way solve all prior hinge problems that worked, plus new ones? i.e. does it disable anything that worked before?

Reply:
This new way doesn't change anything for the user. The user still uses `solve_for_reaction_loads(*reactions)` to calculate reaction forces and does not need to change anything about their input. Not for a beam without hinges nor for a beam with hinges. This can also be seen in the test file. All tests of beams with and without hinges still pass without any changes to the tests. 

For the second question. This new way solves the problem stated in issue #26570 and it keeps all previous functionalities that were there before. But most importantly it adds a new functionality: beams can have multiple hinges. In addition to this hinges can be added easier to a beam than before by using the `apply_hinge` method, without removing the old way of adding hinges by using the `join` method.